cellmlmanip/mathml2sympy/transpiler.py

```diff
@@ -324,112 +274,137 @@ def bvar_handler(node):
     return __wrapped_log
 
 
-def cos_handler():
+def logbase_handler(node):
     """
-    https://www.w3.org/TR/MathML2/chapter4.html#contm.trig
-    unary trigonometric operator
-    """
-    return sympy.cos
+    Qualifier for <log>
 
+    The log function accepts only the logbase schema. If present, the logbase schema denotes the
+    base with respect to which the logarithm is being taken. Otherwise, the log is assumed to be b
+    ase 10. When used with log, the logbase schema is expected to contain a single child schema;
+    otherwise an error is generated.
 
-def tanh_handler():
-    """
-    https://www.w3.org/TR/MathML2/chapter4.html#contm.trig
-    unary trigonometric operator
+    Should be the first element following log, i.e. the second child of the containing apply
+    element.
     """
-    return sympy.tanh
+    return transpile(node)[0]
 
 
-def arccos_handler():
+def simple_operator_handler(tag_name):
     """
-    https://www.w3.org/TR/MathML2/chapter4.html#contm.trig
-    unary trigonometric operator
-    """
-    return sympy.acos
-
+    This function handles simple MathML <tagName> to sympy.Class operators, where no unique handling
+    of tag children etc. is required. The mappings of tagName to sympy class names is in
 
-# CONSTANT AND SYMBOL ELEMENTS #################################################################
-
-def pi_handler():
-    """
-    https://www.w3.org/TR/MathML2/chapter4.html#contm.pi
     """
-    return sympy.pi
+    return getattr(sympy, SIMPLE_MATHML_TO_SYMPY_NAMES[tag_name])
 
 
 # END OF MATHML HANDLERS #######################################################################
 
-# Mapping MathML tag element names (keys) to appropriate function for SymPy output (values)
-HANDLERS = {'abs': abs_handler,
-            'and': and_handler,
-            'apply': apply_handler,
-            'arccos': arccos_handler,
-            'bvar': bvar_handler,
-            'ci': ci_handler,
-            'cn': cn_handler,
-            'cos': cos_handler,
-            'diff': diff_handler,
-            'divide': divide_handler,
-            'eq': eq_handler,
-            'exp': exp_handler,
-            'floor': floor_handler,
-            'geq': geq_handler,
-            'gt': gt_handler,
-            'leq': leq_handler,
-            'ln': ln_handler,
-            'log': log_handler,
-            'lt': lt_handler,
-            'math': math_handler,
-            'minus': minus_handler,
-            'or': or_handler,
-            'otherwise': otherwise_handler,
-            'pi': pi_handler,
-            'piece': piece_handler,
-            'piecewise': piecewise_handler,
-            'plus': plus_handler,
-            'power': power_handler,
-            'rem': rem_handler,
-            'root': root_handler,
-            'tanh': tanh_handler,
-            'times': times_handler}
+# These MathML tags map directly to Sympy class and don't require any extra handling
+SIMPLE_MATHML_TO_SYMPY_NAMES = {
```

Comment:
I've not checked all the Sympy names against their documentation... assuming they're correct!

Reply:
That’s the bit you’re meant to check!! 😉

Reply:
OK!

Reply:
Before you drown in MathML & Sympy documentation…jk!